The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9417138435/job/25941701916) [(plain)](https://github.com/rust-lang/rust/commit/eee85d75c2889140c5354627f7a4019a34e9412c/checks/25941701916/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:a5ac7e51b41094c92402da3b24376905380afc29)
Download action repository 'actions/setup-python@v5' (SHA:82c7e631bb3cdc910f68e0081d67478d79c6982d)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.430   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.447 Collecting boolean-py==4.0
#13 0.453   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.470 Collecting chardet==5.1.0
---
#13 3.537 Building wheels for collected packages: reuse
#13 3.538   Building wheel for reuse (pyproject.toml): started
#13 3.860   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.861   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.861   Stored in directory: /tmp/pip-ephem-wheel-cache-549ns3j0/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.864 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.886   Attempting uninstall: setuptools
#13 3.887     Found existing installation: setuptools 59.6.0
#13 3.888     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
#13 5.107   Downloading virtualenv-20.26.2-py3-none-any.whl (3.9 MB)
#13 5.273      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 24.0 MB/s eta 0:00:00
#13 5.326 Collecting filelock<4,>=3.12.2
#13 5.332   Downloading filelock-3.14.0-py3-none-any.whl (12 kB)
#13 5.365 Collecting platformdirs<5,>=3.9.1
#13 5.371   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.392 Collecting distlib<1,>=0.3.7
#13 5.399   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.410      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 52.4 MB/s eta 0:00:00
#13 5.495 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.655 Successfully installed distlib-0.3.8 filelock-3.14.0 platformdirs-4.2.2 virtualenv-20.26.2
#13 DONE 5.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      200640 kB
DirectMap2M:     8187904 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/1be24d70ced0d6b8d41a48b6a28b3790f6facf4c/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-1be24d70ced0d6b8d41a48b6a28b3790f6facf4c-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 25.98s
##[endgroup]
fmt check
##[error]Diff in /checkout/tests/codegen/slice-pointer-nonnull-unwrap.rs at line 9:
 // CHECK-NEXT: ret i64 %ptr.1
 #[no_mangle]
 pub fn slice_ptr_len_1(ptr: *const [u8]) -> usize {
-  let ptr = ptr.cast_mut();
-  if let Some(ptr) = NonNull::new(ptr) {
-    ptr.len()
-  } else {
-    // We know ptr is null, so we know ptr.wrapping_byte_add(1) is not null.
-    NonNull::new(ptr.wrapping_byte_add(1)).unwrap().len()
+    let ptr = ptr.cast_mut();
+    if let Some(ptr) = NonNull::new(ptr) {
+        ptr.len()
+    } else {
+    } else {
+        // We know ptr is null, so we know ptr.wrapping_byte_add(1) is not null.
+        NonNull::new(ptr.wrapping_byte_add(1)).unwrap().len()
 }
 
 
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/sys/pal/xous/mod.rs" "/checkout/library/std/src/sys/pal/xous/stdio.rs" "/checkout/tests/codegen/gdb_debug_script_load.rs" "/checkout/tests/codegen/atomic-operations.rs" "/checkout/tests/codegen/virtual-function-elimination.rs" "/checkout/tests/codegen/slice-as_chunks.rs" "/checkout/tests/codegen/powerpc64le-struct-align-128.rs" "/checkout/tests/codegen/generic-debug.rs" "/checkout/tests/codegen/is_val_statically_known.rs" "/checkout/tests/codegen/float/f128.rs" "/checkout/tests/codegen/float/f16.rs" "/checkout/tests/codegen/debug-accessibility/crate-enum.rs" "/checkout/tests/codegen/debug-accessibility/struct-fields.rs" "/checkout/tests/codegen/debug-accessibility/private-enum.rs" "/checkout/tests/codegen/debug-accessibility/public-struct.rs" "/checkout/tests/codegen/debug-accessibility/private-struct.rs" "/checkout/tests/codegen/debug-accessibility/crate-struct.rs" "/checkout/tests/codegen/debug-accessibility/tuple-fields.rs" "/checkout/tests/codegen/debug-accessibility/super-enum.rs" "/checkout/tests/codegen/debug-accessibility/super-struct.rs" "/checkout/tests/codegen/debug-accessibility/public-enum.rs" "/checkout/tests/codegen/panic-in-drop-abort.rs" "/checkout/tests/codegen/ptr-arithmetic.rs" "/checkout/tests/codegen/naked-fn/naked-noinline.rs" "/checkout/tests/codegen/naked-fn/naked-functions.rs" "/checkout/tests/codegen/naked-fn/naked-nocoverage.rs" "/checkout/tests/codegen/remap_path_prefix/aux_mod.rs" "/checkout/tests/codegen/remap_path_prefix/main.rs" "/checkout/tests/codegen/remap_path_prefix/xcrate-generic.rs" "/checkout/tests/codegen/remap_path_prefix/auxiliary/xcrate-generic.rs" "/checkout/tests/codegen/remap_path_prefix/auxiliary/remap_path_prefix_aux.rs" "/checkout/tests/codegen/remap_path_prefix/issue-73167-remap-std.rs" "/checkout/tests/codegen/asm-sanitize-llvm.rs" "/checkout/tests/codegen/checked_math.rs" "/checkout/tests/codegen/debug-column.rs" "/checkout/tests/codegen/some-global-nonnull.rs" "/checkout/tests/codegen/bool-cmp.rs" "/checkout/tests/codegen/slice-position-bounds-check.rs" "/checkout/tests/codegen/vec-optimizes-away.rs" "/checkout/tests/codegen/no_builtins-at-crate.rs" "/checkout/tests/codegen/no-jump-tables.rs" "/checkout/tests/codegen/vec_pop_push_noop.rs" "/checkout/tests/codegen/inline-always-works-always.rs" "/checkout/tests/codegen/vec-in-place.rs" "/checkout/tests/codegen/integer-overflow.rs" "/checkout/tests/codegen/drop-in-place-noalias.rs" "/checkout/tests/codegen/async-fn-debug-awaitee-field.rs" "/checkout/tests/codegen/asm-clobbers.rs" "/checkout/tests/codegen/sroa-fragment-debuginfo.rs" "/checkout/tests/codegen/option-niche-eq.rs" "/checkout/tests/codegen/slice-pointer-nonnull-unwrap.rs" "/checkout/tests/codegen/stack-protector.rs" "/checkout/tests/codegen/debuginfo-generic-closure-env-names.rs" "/checkout/tests/codegen/simd/packed-simd-alignment.rs" "/checkout/tests/codegen/simd/simd-wide-sum.rs" "/checkout/tests/codegen/simd/issue-120720-reduce-nan.rs" "/checkout/tests/codegen/simd/packed-simd.rs" "/checkout/tests/codegen/simd/swap-simd-types.rs" "/checkout/tests/codegen/simd/simd_arith_offset.rs" "/checkout/tests/codegen/simd/unpadded-simd.rs" "/checkout/tests/codegen/noalias-flag.rs" "/checkout/tests/codegen/dont_codegen_private_const_fn_only_used_in_const_eval.rs" "/checkout/tests/codegen/mem-replace-big-type.rs" "/checkout/tests/codegen/align-byval.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Fri Jun  7 12:45:27 UTC 2024
  network time: Fri, 07 Jun 2024 12:45:28 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/tests/assembly/issue-83585-small-pod-struct-equality.rs
+++ b/tests/assembly/issue-83585-small-pod-struct-equality.rs
@@ -1,0 +1,30 @@
+//@ assembly-output: emit-asm
+//@ compile-flags:-Copt-level=3
+//@ only-x86_64
+
+#![crate_type = "lib"]
+
+#[no_mangle]
+type T = u8;
+type T1 = (T, T, T, T, T, T, T, T);
+type T2 = [T; 8];
+
+#[no_mangle]
+// CHECK-LABEL: foo1a
+// CHECK: cmp
+// CHECK-NEXT: set
+// CHECK-NEXT: ret
+pub fn foo1a(a: T1, b: T1) -> bool {
+    a == b
+}
+
+#[no_mangle]
+// CHECK-LABEL: foo1b
+// CHECK: mov
+// CHECK-NEXT: cmp
+// CHECK-NEXT: set
+// CHECK-NEXT: ret
+pub fn foo1b(a: &T1, b: &T1) -> bool {
+    a == b
+}
+

--- a/tests/codegen/issues/issue-111508-vec-tryinto-array.rs
+++ b/tests/codegen/issues/issue-111508-vec-tryinto-array.rs
@@ -1,0 +1,22 @@
+//@ compile-flags: -O
+// This regress since Rust version 1.72.
+//@ min-llvm-version: 18.1.4
+
+#![crate_type = "lib"]
+
+use std::convert::TryInto;
+
+const N: usize = 24;
+
+#[no_mangle]
+// CHECK-LABEL: @example
+// CHECK-NOT: unwrap_failed
+pub fn example(a: Vec<u8>) -> u8 {
+    if a.len() != 32 {
+        return 0;
+    }
+
+    let a: [u8; 32] = a.try_into().unwrap();
+
+    a[15] + a[N]
+}

--- a/tests/codegen/slice-pointer-nonnull-unwrap.rs
+++ b/tests/codegen/slice-pointer-nonnull-unwrap.rs
@@ -1,0 +1,19 @@
+//@ compile-flags: -O
+//@ min-llvm-version: 18
+#![crate_type = "lib"]
+
+use std::ptr::NonNull;
+
+// CHECK-LABEL: @slice_ptr_len_1
+// CHECK: {{.*}}:
+// CHECK-NEXT: ret i64 %ptr.1
+#[no_mangle]
+pub fn slice_ptr_len_1(ptr: *const [u8]) -> usize {
+  let ptr = ptr.cast_mut();
+  if let Some(ptr) = NonNull::new(ptr) {
+    ptr.len()
+  } else {
+    // We know ptr is null, so we know ptr.wrapping_byte_add(1) is not null.
+    NonNull::new(ptr.wrapping_byte_add(1)).unwrap().len()
+  }
+}

